packages/blade/src/components/Dropdown/_decisions/controlled-dropdown.md

```diff
@@ -0,0 +1,342 @@
+        // New Prop 👇🏼
+        isSelected={currentSelection === 'mumbai'}
+        title="Mumbai"
+        value="mumbai"
+      />
+      <ActionListItem
+        isSelected={currentSelection === 'bangalore'}
+        title="Bangalore"
+        value="bangalore"
+      />
+    </ActionList>
+  </DropdownOverlay>
+</Dropdown>;
+```
+
+</details>
+
+<details>
+<summary><code>onChange</code> + <code>value</code></summary>
+
+#### `onChange` + `value`
+
+```jsx
+const [currentSelection, setCurrentSelection] = React.useState();
+
+const onSomeAction = () => {
+  setCurrentSelection('mumbai');
+};
+
+const onResetClick = () => {
+  setCurrentSelection(undefined);
+};
+
+const onSelectChange = ({ values }) => {
+  setCurrentSelection(values[0]);
+};
+
+<Dropdown>
+  <SelectInput
+    label="City"
+    // New Prop 👇🏼
+    value={currentSelection}
+    onChange={onSelectChange}
+  />
+  <DropdownOverlay>
+    <ActionList>
+      <ActionListItem title="Mumbai" value="mumbai" />
+      <ActionListItem title="Bangalore" value="bangalore" />
+    </ActionList>
+  </DropdownOverlay>
+</Dropdown>;
+```
+
+</details>
+
+<details>
+<summary><code>useSelect</code> hook</summary>
+
+#### useSelect
+
+```jsx
+const { selectItem, reset, dropdownRef } = useSelect();
+
+const onSomeAction = () => {
+  selectItem('mumbai');
+};
+
+const onResetClick = () => {
+  reset();
+};
+
+<Dropdown ref={dropdownRef}>
+  <SelectInput />
+  <DropdownOverlay>
+    <ActionList>
+      <ActionListItem title="Mumbai" value="mumbai" />
+      <ActionListItem title="Bangalore" value="bangalore" />
+    </ActionList>
+  </DropdownOverlay>
+</Dropdown>;
+```
+
+Inspirations
+
+</details>
+
+## Concluded Solution
+
+Based on the [discussions](#approach-comparisons), we decided to go with a combination of approaches propsed above-
+
+- On `SelectInput`, `value` + `onChange` + `defaultValue`
+- On `ActionListItem`, `isSelected` + `onClick` + `isDefaultSelected`
+
+### When Trigger is `SelectInput`
+
+For SelectInput (and maybe AutoComplete in future), we decided to go with `value` + `onChange` props on the trigger. This keeps the controlled state props on the same component as opposed to `isSelected` + `onChange` approach where `isSelected` goes on ActionListItem and `onChange` goes on SelectInput.
+
+```jsx
+const [currentSelection, setCurrentSelection] = React.useState();
+
+const onSomeAction = () => {
+  setCurrentSelection('mumbai');
+};
+
+const onResetClick = () => {
+  setCurrentSelection(undefined);
+};
+
+const onSelectChange = ({ values }) => {
+  setCurrentSelection(values[0]);
+};
+
+<Dropdown>
+  <SelectInput
+    label="City"
+    // New Prop 👇🏼
+    value={currentSelection}
+    onChange={onSelectChange}
+  />
+  <DropdownOverlay>
+    <ActionList>
+      <ActionListItem title="Mumbai" value="mumbai" />
+      <ActionListItem title="Bangalore" value="bangalore" />
+    </ActionList>
+  </DropdownOverlay>
+</Dropdown>;
+```
+
+### When Trigger is Button or other triggers
```

Comment:
add another example where ActionList is used outside of DropDown? For eg: BottomSheet

Reply:
It is based on trigger and not necessarily based on what it is being rendered inside. E.g. even in SelectInput BottomSheet, we do use `Dropdown` component so the same `onChange` + `value` will continue to work.

And in BottomSheet with Button, the same Button trigger rules apply so the same example is valid for that as well. 

It won't really change anything in the approach so didn't add example here to not deviate and confuse from the focus of this part.